src/AvTranscoder/properties/StreamProperties.cpp

```diff
@@ -139,8 +139,12 @@ size_t StreamProperties::getBitRate() const
             if(positionOfFirstKeyFrame != -1 && positionOfLastKeyFrame != -1)
                 break;
         }
+        // Close a given AVCodecContext and free all the data associated with it (but not the AVCodecContext itself)
         avcodec_close(_codecContext);
 
+        // Returns at the beginning of the stream
+        const_cast<FormatContext*>(&_fileProperties->getFormatContext())->seek(0, AVSEEK_FLAG_BYTE);
```

Comment:
So, here you go back to the beginning of the file, while you might be in the middle of it just before the analyze (after a seek, for instance). Perhaps you could force the analyze to start at the 0 position in the file, or just go back to the previous position after the analyze. Maybe the second solution offers more flexibility to the user, don't you think?